### ISSUE TITLE
[pyreverse] Attributes for nodes set by Linker can't be type hinted 

### ISSUE THREAD

**Issue by qequ** [PyCQA/pylint#7850]
### Bug description

when the `Linker` works over astroid nodes and set attributes for them , e.g.  when setting `instance_attrs_type`

```python      
  node.instance_attrs_type = collections.defaultdict(list)
```
when trying to type hint this attributes like

```python      
  node.instance_attrs_type: defaultdict[str, list[...]] = collections.defaultdict(list)
```

mypy throws an error like this

```
Type cannot be declared in assignment to non-self attribute [misc]
```

CC: @DudeNr33 

### Command used

```shell
tested with mypy on pre-commit
```

### Pylint output

```shell
(no pylint related)
```

### Expected behavior

being able to type hint new attributes for the nodes

### Pylint version

```shell
(tested with source code)
```

### OS / Environment

ubuntu 20.04

**Comment by DanielNoord** [comment]
As said by @DudeNr33 this should be fixed by using dataclass wrappers around nodes instead of changing the nodes directly. I'm changing the labels and name of the issue.
